src/build.rs

```diff
@@ -40,6 +49,12 @@ pub struct CheckoutBuilder<'cb> {
 /// of completed steps so far, and the final is the total number of steps.
 pub type Progress<'a> = FnMut(Option<&Path>, usize, usize) + 'a;
 
+/// Checkout notifications callback.
+///
+/// The first argument is the notification type, the next is the path for the
+/// the notification, followed by the baseline diff, target diff, and workdir diff.
+pub type Notify<'a> = FnMut(CheckoutNotificationType, Option<&Path>, DiffFile, DiffFile, DiffFile) -> c_int + 'a;
```

Comment:
Could this return a `bool` to be a bit more "rust-y"? We could translate that to zero/nonzero under the hood